src/odemis/util/dataio.py

```diff
@@ -319,6 +320,17 @@ def leader_quality(da):
         if not das:
             raise ValueError(f"No compatible 2D data found in file {fn}")
 
+        # handle scan rotation
+        dar = []
+        for da in das:
+            scan_rot = da.metadata.get(model.MD_BEAM_SCAN_ROTATION, 0)
+            if math.isclose(scan_rot, math.pi, abs_tol=0.1):
+                # rotate image content (Note: assumption is only scan rotation 180 is used.)
+                logging.debug(f"Found beam scan rotation ({math.degrees(scan_rot)}) rotating image content")
+                da = numpy.rot90(da, 2)
+            dar.append(da)
```

Comment:
Can we add a note to flag rotation angle other than pi, and log warning that the stitching might be influenced due to that?